### ISSUE TITLE
[Awaiting C+ payment] [HOLD for payment 2023-07-24] [$1000] Phone number contact in task assignee is displayed with @expensify.sms

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#22050]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Click plus and click on assign task
3. Enter any title and click on next
4. Click on Assignee
5. Search for user with phone number as default contact method and select it
6. Observe that on new task page under assignee, the phone number contact is displayed with @expensify.sms in the end
## Expected Result: 
App should display phone number contact below assignee in new format as it displays in assignee search
## Actual Result: 
App displays phone number contact in old format and with @expensify.sms behind it below assignee

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop
 
**Version Number:** 1.3.35-5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/a549f5c3-bedf-465e-9ad0-7f8d69043df1

https://github.com/Expensify/App/assets/43996225/59ff6ee6-6ff9-4214-bb12-838412f517e2

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688140198815779

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010ead72a986dba400</li>
        <li>Upwork Job ID: 1676235816963620864</li>
        <li>Last Price Increase: 2023-07-04</li>
    </ul>
</details>

**Comment by trjExpensify** [comment]
Yup, this is reproducible. As we do elsewhere throughout the app, we shouldn't be showing the `@expensify.sms` portion of the phone number login. Let's not reinvent the wheel and extend however we're doing that elsewhere here on the taskReport. 

@thienlnam putting this on your radar as well, as it relates to tasks. Going to move it `External` for the time being.

**Comment by Adhiccup** [comment]
There are several approach.
split phone number in frontside,
and make function such like FormatPhoneNumber() , and fix api(BE)
or remake <MenuItem> component.

**Comment by Adhiccup** [comment]
Contributor details
my Expensify account email: admongd99@gamil.com
Upwork Profile Link: https://www.upwork.com/freelancers/~017b2ac34ea29e2afb?viewMode=1

but my upwork account is not good because my last account has problem  so i create it today.

**Comment by Adhiccup** [comment]
yes,  ok,  I'll  contact after fix it.

**Comment by trjExpensify** [comment]
>@trjExpensify just a note that we have to start a discussion on slack (after this is fixed) on how to prevent @expensify.sms bugs, because they keep happening.

Yeah, I'm not sure why they are all of a sudden creeping in again, but I agree it's a good discussion on the checklist item to prevent them moving forward.

**Comment by trjExpensify** [comment]
Are we all cleared up here, when can we expect the PR to be linked? Thanks!

**Comment by trjExpensify** [comment]
We require a 🟢 from @bondydaa actually. :)

**Comment by bondydaa** [comment]
oh wait maybe I remembered incorrectly, this bit of code in the API seems to suggest we do expect the `@expensify.sms` from clients https://github.com/Expensify/Web-Expensify/blob/313880bfc5062e846d75bbe0ee0fda28201a3925/api.php#L109-L119.

**Comment by bondydaa** [comment]
asked in slack https://expensify.slack.com/archives/C03TQ48KC/p1689005020724049

**Comment by bondydaa** [comment]
yes both of those links were internal only

**Comment by bondydaa** [comment]
sorry, looks like I had it backwards and we do expect clients to strip out the sms domain from the logins 👍

**Comment by bondydaa** [comment]
approved the PR 👍

**Comment by rushatgabhane** [comment]
reviewing today

**Comment by kbecciv** [comment]
@trjExpensify @bondydaa I can reproduce above issue on build 1.3.42-21

https://github.com/Expensify/App/assets/93399543/214633ed-4f09-4fb2-b973-a5f0ef64b07b

![image (35)](https://github.com/Expensify/App/assets/93399543/c850bd7c-aca2-4120-aafb-ed33455d87ae)

**Comment by trjExpensify** [comment]
>In the video uploaded just above, the number is 8 (705) XXXX but the email is +7705. Why is the first digit changing from 8 to 7?

Not sure, maybe it's something to do with the phone number validation - I would expect to see the country code there as well. @kbecciv what country code was used/is this a US number? Also, can you retest this issue with `expensify.sms` showing up on the task row in the chat. Thanks!

**Comment by trjExpensify** [comment]
Yeah, I'm happy to go from when Bondy confirmed the green light to proceed after the clarification, so we'll apply the #urgency bonus here! 

Payments due by my calculation to confirm are:

- @dhanashree-sawant - $250 for the bug report
- @c3024 - $1,500 for the fix & #urgency bonus 
- @rushatgabhane - $1,500 for C+ & #urgency bonus 

I'll make sure @c3024 & @dhanashree-sawant have offers in Upwork. @rushatgabhane can you complete the checklist items though please, so we can proceed with payment? As we discussed earlier, we want to try and limit these `@expensify.sms` bugs from creeping in time and time again. Thanks! 

>In the video uploaded just above, the number is 8 (705) XXXX but the email is +7705. Why is the first digit changing from 8 to 7?

As for this, it's unrelated to this issue so @c3024 can you investigate it a bit more and if it's not reported yet, pop that one through #expensify-bugs as per the process as you found it. Thanks!

**Comment by JmillsExpensify** [comment]
Reviewed details for @rushatgabhane. This is accurate and approved for payment in NewDot.

**Comment by trjExpensify** [comment]
Cool, awaiting the checklist!

**Comment by trjExpensify** [comment]
Yes, we typically don't proceed with payments until the checklist is complete.

**Comment by rushatgabhane** [comment]
WIP ⚠️ 

1. The PR that introduced the bug has been identified. Link to the PR:

2. The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

3. A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

4. Determine if we should create a regression test for this bug.

5. If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again

**Comment by rushatgabhane** [comment]
Created a manual request https://staging.new.expensify.com/r/8107948973342401

**Comment by trjExpensify** [comment]
Thanks, @rushatgabhane! 

@dhanashree-sawant - settled up. 
@c3024 I was just about to settle up, but I can see you've somehow already been paid $1,000 for this job. I've sent a bonus of $500 but can you confirm? 😕

**Comment by rushatgabhane** [comment]
@trjExpensify you can close this issue because creating a request on new dot is enough. it'll be tracked there

**Comment by trjExpensify** [comment]
I'm seeking confirmation on that from @JmillsExpensify elsewhere actually.

**Comment by rushatgabhane** [comment]
I've been paid. we can close 😅

**Comment by trjExpensify** [comment]
Dope!

**Comment by bondydaa** [comment]
as reported here https://expensify.slack.com/archives/C049HHMV9SM/p1692114818807249

looks like either this wasn't fully fixed by https://github.com/Expensify/App/pull/22566 or there is a new regression

<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/5e571af3-d927-4457-947a-2ac04e7b9239">

quickly looking at the PR I don't see any screenshots or videos that show the task preview so we might have just missed it 🤦 

Going to spin up a new issue for this.

**Comment by bondydaa** [comment]
ah thanks for that context! if you can find which PR might have caused it then great! we can tie it to that but otherwise we can get it handled on the new GH that gets spun up.

**Comment by kbecciv** [comment]
Yes, we have another issue reported by Contributor
Slack conversation: https://expensify.slack.com/archives/C049HHMV9SM/p1694333604061429

Uploading Screen Recording 2023-08-15 at 22.54.24.mov…

.

**Comment by trjExpensify** [comment]
>Going to spin up a new issue for this.

@bondydaa where is that? Let's not reopen this one that's already done and dusted with payments made.

**Comment by kbecciv** [comment]
Created a new issue here https://github.com/Expensify/App/issues/27330, closing old one.
